### PR TITLE
core/sigagg: verify aggregated signature

### PR DESCRIPTION
**Pull request by xenowits**
Verify reconstructed aggregated signature.

Fixes OBOL-17 of sigp audit.

category: feature 
ticket: #2122

### CODE REVIEW

File: core/sigagg/sigagg.go [hunk 1]
```diff
@@ -89,3 +95,25 @@ func (a *Aggregator) Aggregate(ctx context.Context, duty core.Duty, pubkey core.
...
+	}
+
+	msgRoot, err := sigData.MessageRoot()
+	if err != nil {
+		return errors.Wrap(err, "message root")
+	}
+
+	aggSig := tbls.Signature(sigData.Signature())
+
+	err = tbls.Verify(tblsPubkey, msgRoot[:], aggSig)
```
Comment (corverroos) [inline core/sigagg/sigagg.go:113]: this is a eth2 signature, so it should include eth2 domains. I do not want to add a eth2client to sigagg, so I would actually mark this as WONTFIX.

Reasoning:
  - We already validate the partial signatures.
  - So if aggregation succeeds then resulting aggregate signature is also valid. 
  - Additional validation isn't required.

**Review by corverroos — CHANGES_REQUESTED** [review]

File: core/sigagg/sigagg.go (hunk 1 above)
Comment (corverroos) [reply core/sigagg/sigagg.go:113]: If invalid aggregate signatures are however still possible given valid partial signatures and succesful aggregation, then we should follow the parsigex approach of decoupling validation from the component via a `verifyFunc`.

Comment (corverroos) [reply core/sigagg/sigagg.go:113]: See parsigex how to verify eth2 signatures

Comment (xenowits) [reply core/sigagg/sigagg.go:113]: I would say we err on the side of caution here and still validate aggregated signatures. I'll follow the `verifyFunc` approach of verifying signatures

File: core/sigagg/sigagg.go [hunk 2]
```diff
@@ -89,3 +100,25 @@ func (a *Aggregator) Aggregate(ctx context.Context, duty core.Duty, pubkey core.
 
 	return nil
 }
+
+// NewSigAggVerifier returns a signature verification function for aggregated signatures.
+func NewSigAggVerifier(eth2Cl eth2wrap.Client) func(context.Context, core.PubKey, core.SignedData) error {
```
Comment (corverroos) [inline core/sigagg/sigagg.go:105]: avoid stutter, suggest `NewEth2Verifier` or just `NewVerifier` since there isn't a second implementation

**Review by corverroos — APPROVED** [review]

**Pull request merged**